.github/workflows/ca-publishing-crl-file-test.yml

```diff
@@ -61,15 +61,71 @@ jobs:
+          docker exec pki sed -i \
+              -e "s/^\(policyset.userCertSet.list\)=\(.*\)$/\1=\2,11/" \
+              -e "$ a policyset.userCertSet.11.constraint.class_id=noConstraintImpl" \
+              -e "$ a policyset.userCertSet.11.constraint.name=No Constraint" \
+              -e "$ a policyset.userCertSet.11.default.class_id=crlDistributionPointsExtDefaultImpl" \
+              -e "$ a policyset.userCertSet.11.default.name=CRL Distribution Points Extension Default" \
+              -e "$ a policyset.userCertSet.11.default.params.crlDistPointsCritical=false" \
+              -e "$ a policyset.userCertSet.11.default.params.crlDistPointsNum=1" \
+              -e "$ a policyset.userCertSet.11.default.params.crlDistPointsEnable_0=true" \
+              -e "$ a policyset.userCertSet.11.default.params.crlDistPointsIssuerName_0=cn=CA Signing Certificate,ou=pki-tomcat,o=EXAMPLE" \
+              -e "$ a policyset.userCertSet.11.default.params.crlDistPointsIssuerType_0=DirectoryName" \
+              -e "$ a policyset.userCertSet.11.default.params.crlDistPointsPointName_0=$URI" \
+              -e "$ a policyset.userCertSet.11.default.params.crlDistPointsPointType_0=URIName" \
+              -e "$ a policyset.userCertSet.11.default.params.crlDistPointsReasons_0=" \
+              /var/lib/pki/pki-tomcat/conf/ca/profiles/ca/caUserCert.cfg
+
+          # check updated profile
+          docker exec pki cat /var/lib/pki/pki-tomcat/conf/ca/profiles/ca/caUserCert.cfg
+
+      - name: Configure caServerCert profile
+        run: |
+          # remove AIA extension
+          docker exec pki sed -i \
+              -e "s/^\(policyset.serverCertSet.list\)=.*$/\1=1,2,3,4,6,7,8,12/" \
+              -e "/^policyset.serverCertSet.5/d" \
+              /var/lib/pki/pki-tomcat/conf/ca/profiles/ca/caServerCert.cfg
+
+          # check updated profile
+          docker exec pki cat /var/lib/pki/pki-tomcat/conf/ca/profiles/ca/caServerCert.cfg
+
+      - name: Prepare CRL publishing location
         run: |
           # create CRL folder
           docker exec pki mkdir -p /var/lib/pki/pki-tomcat/crl
-          docker exec pki chown pkiuser.pkiuser /var/lib/pki/pki-tomcat/crl
+          docker exec pki chown -R pkiuser:pkiuser /var/lib/pki/pki-tomcat/crl
+
+          # create CRL webapp config
+          # use allowLinking=true since MasterCRL.crl is a link
+          # use cachingAllowed=false since MasterCRL.crl is not static
+          cat > crl.xml << EOF
+          <Context docBase="/var/lib/pki/pki-tomcat/crl">
+              <Resources allowLinking="true" cachingAllowed="false" />
+          </Context>
+          EOF
+
+          # deploy CRL webapp
+          docker cp crl.xml pki:/var/lib/pki/pki-tomcat/conf/Catalina/localhost
+          docker exec pki chown -R pkiuser:pkiuser /var/lib/pki/pki-tomcat/conf/Catalina/localhost/crl.xml
+          docker exec pki ls -l /var/lib/pki/pki-tomcat/conf/Catalina/localhost
```

Comment:
If we modify the URI to `http://pki.example.com:8080/ca/crl/MasterCRL.crl` then the folder could be associated with the ca web app and no need for a new webapp. This should be easier for future evolution in case this configuration has to be generated during CA installation with `pkispawn`

Reply:
By default the CA webapp is deployed from `/usr/share/pki/ca/webapps/ca` which is owned by the RPM package and not supposed to be modified. If we want to publish the CRL under CA webapp we probably need to copy CA webapp files into a new folder, then create a context file pointing to the new folder, so it's actually more complicated than creating a separate CRL webapp. Here the CRL webapp is only created for testing, but we probably should investigate and document the recommended way to publish the CRL in Tomcat.